native/src/modules/endpoint/hocs/withPayloadProvider.js

```diff
@@ -41,31 +41,33 @@ export type SuccessType<S: {}, R: {}> = {|
   refreshProps: R
 |}
 
-export type StatusPropsType<S: {}, R: {}> =
+export type StatusPropsType<S: { dispatch: Dispatch<StoreActionType> }, R: {}> =
   RouteNotInitializedType
   | LoadingType<$Diff<S, { dispatch: Dispatch<StoreActionType> }>, R>
   | ErrorType<R>
   | LanguageNotAvailableType
   | SuccessType<$Diff<S, { dispatch: Dispatch<StoreActionType> }>, R>
 
-export type PropsType<S: { dispatch: Dispatch<StoreActionType> }, R: {}> = {|
+export type PropsType<S: { dispatch: Dispatch<StoreActionType> }, R: {}, T: RoutesType> = {|
   ...StatusPropsType<S, R>,
   dispatch: Dispatch<StoreActionType>,
-  navigation: NavigationScreenProp<*>,
+  navigation: NavigationPropType<T>,
+  route: RoutePropType<T>,
   t?: TFunction
-|} | {|
+|} | {| // Necessary because of weird flow error saying t is missing in containers
```

Comment:
I dont understand this comment. I think it can be removed.

The types here could be extracted to type variables `const type Type = ...` though for better readability.

Reply:
It is already specified as separate type.
The comment means that 
```
{|
  ...StatusPropsType<S, R>,
  dispatch: Dispatch<StoreActionType>,
  navigation: NavigationPropType<T>,
  route: RoutePropType<T>
|}
```
should already be handled by the type above since the only difference is an optional `t` which means there does not have to be one:
```
{|
  ...StatusPropsType<S, R>,
  dispatch: Dispatch<StoreActionType>,
  navigation: NavigationPropType<T>,
  route: RoutePropType<T>,
  t?: TFunction
|}```

Reply:
Changed the comment to ` // Necessary because of weird flow error saying t is missing even though t is already optional in the first type`